iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions2.cpp

```diff
@@ -49,13 +49,31 @@ static void insertDynamicShapeDimOperands(DispatchWorkgroupsOp regionOp,
   // NOTE: this may insert new shape values at |builder|, which is prior to our
   // dispatch operation. All new values that are built can only depend on SSA
   // values that are defined prior to the region op.
+  Location loc = regionOp.getLoc();
   auto shapeValue = Shape::buildOrFindRankedShapeForValue(
-      regionOp.getLoc(), value, builder.getIndexType(), builder);
+      loc, value, builder.getIndexType(), builder);
+  unsigned indexOfDynamicDim = 0;
   for (int dim = 0, e = shapedType.getRank(); dim < e; ++dim) {
-    if (shapedType.isDynamicDim(dim)) {
-      newOperands.push_back(builder.create<Shape::RankedDimOp>(
-          regionOp.getLoc(), shapeValue, dim));
+    if (!shapedType.isDynamicDim(dim)) {
+      ++indexOfDynamicDim;
+      continue;
     }
+    auto makeRankedShapeOp =
```

Comment:
why can't you use RankedDimOp? it's what prevents the need for this traversal

Reply:
Ah thanks, it was because I needed a `tie_shape` in DispatchRegion to talk properly to OutlineDispatchRegions2 but I created all `dims/shape/tie_shape` after the `dispatch.workgroup` which created dominance issues that this was trying to fix.

Creating `dims/shape` before the `dispatch.workgroup` and the `tie_shape` after, makes all this unnecessary.

Reply:
Turns out this seems to also be the cause of my off by one issue, so double thanks! :)